java/client/src/org/openqa/selenium/edge/ChromiumEdgeDriverService.java

```diff
@@ -0,0 +1,182 @@
+
+  /**
+   * System property that defines comma-separated list of remote IPv4 addresses which are
+   * allowed to connect to MSEdgeDriver.
+   */
+  public final static String EDGE_DRIVER_ALLOWED_IPS_PROPERTY = "webdriver.edge.withAllowedIps";
+
+  public ChromiumEdgeDriverService(
+      File executable,
+      int port,
+      List<String> args,
+      Map<String, String> environment) throws IOException {
+    super(executable, port, args, environment);
+  }
+
+  /**
+   * Configures and returns a new {@link ChromiumEdgeDriverService} using the default configuration. In
+   * this configuration, the service will use the MSEdgeDriver executable identified by the
+   * {@link #EDGE_DRIVER_EXE_PROPERTY} system property. Each service created by this method will
+   * be configured to use a free port on the current system.
+   *
+   * @return A new ChromiumEdgeDriverService using the default configuration.
+   */
+  public static ChromiumEdgeDriverService createDefaultService() {
+    return new ChromiumEdgeDriverService.Builder().build();
+  }
+
+  /**
+   * Builder used to configure new {@link ChromiumEdgeDriverService} instances.
+   */
+  @AutoService(DriverService.Builder.class)
+  public static class Builder extends EdgeDriverService.Builder<
+      ChromiumEdgeDriverService, ChromiumEdgeDriverService.Builder> {
+
+    private boolean verbose = Boolean.getBoolean(EDGE_DRIVER_VERBOSE_LOG_PROPERTY);
+    private boolean silent = Boolean.getBoolean(EDGE_DRIVER_SILENT_OUTPUT_PROPERTY);
+    private String withAllowedIps = System.getProperty(EDGE_DRIVER_ALLOWED_IPS_PROPERTY);
+
+    @Override
+    public boolean isLegacy() {
+      return false;
+    }
+
+    @Override
+    public int score(Capabilities capabilities) {
+      int score = 0;
+
+      if (BrowserType.EDGE.equals(capabilities.getBrowserName())) {
+        score++;
+      }
+
+      if (capabilities.getCapability(EdgeOptions.CAPABILITY) != null) {
+        score++;
+      }
+
+      return score;
+    }
+
+    /**
+     * Configures the driver server verbosity.
+     *
+     * @param verbose whether verbose output is used
+     * @return A self reference.
+     */
+    @Override
+    public EdgeDriverService.Builder withVerbose(boolean verbose) {
+      this.verbose = verbose;
+      return this;
+    }
+
+    /**
+     * Configures the driver server for silent output.
+     *
+     * @param silent whether silent output is used
+     * @return A self reference.
+     */
+    public ChromiumEdgeDriverService.Builder withSilent(boolean silent) {
+      this.silent = silent;
+      return this;
+    }
+
+    /**
+     * Configures the comma-separated list of remote IPv4 addresses which are allowed to connect
+     * to the driver server.
+     *
+     * @param withAllowedIps Comma-separated list of remote IPv4 addresses.
+     * @return A self reference.
+     */
+    public ChromiumEdgeDriverService.Builder withWhitelistedIps(String withAllowedIps) {
```

Comment:
At some point, we need to change this language from "white listing" to "allow list". We can do that in a follow up PR.